### PR TITLE
Reduce the size of the verifier for schemes G16 and PGHR13

### PR DESCRIPTION
**Pull request by m1cm1c**
Reduces the size of the verifier for schemes G16 and PGHR13 by 400 LOC (e.g. from 595 LOC to 195 LOC for a small proof).

### CODE REVIEW

**Comment by Schaeff** [comment]
Good idea! Can you run `cargo fmt` on your code to hopefully fix the tests?

**Comment by m1cm1c** [comment]
@Schaeff Done.

**Review by dark64 — DISMISSED** [review]

**Comment by Schaeff** [comment]
I think it would be worth refactoring this a bit and introducing a function `fn solidity_pairing_lib(with_g2_addition: bool) -> String` which introduces the G2 addition library and the G2 addition function in the `Pairing` library if `with_g2_addition` is true. This would avoid duplication of this solidity code.

**Comment by m1cm1c** [comment]
I now included #1009 into this PR as it does not make sense anymore to keep it separate.

**Comment by Schaeff** [comment]
Seems like the formatting issue should be resolved by running `cargo fmt` again

**Comment by m1cm1c** [comment]
Weird. I thought I ran it before committing. Sorry.

**Comment by Schaeff** [comment]
No worries! We have a git hook which makes sure `cargo fmt` passes before you commit.
Now clippy is complaining, you can run `cargo clippy` locally and see what it's unhappy with, these are small changes.

**Comment by m1cm1c** [comment]
It's going to be fewer commits next time. It's my first time I did anything with rust.

**Comment by Schaeff** [comment]
No worries at all. Something needs to be updated in the integration tests, let me know if I should have a look at it.

**Comment by m1cm1c** [comment]
>let me know if I should have a look at it

Yes, would be nice. I cannot view the output of the integration test.

**Comment by Schaeff** [comment]
Hey @m1cm1c I tried pushing to your branch but seems like I don't have the permission. I think the issue is that gm17 should have `true` for the G2 lib, currently it's `false`. You can make the change yourself if you want and add a changelog entry.

**Comment by m1cm1c** [comment]
@Schaeff Both you and @dark64 now have push access (role "maintain").

**Comment by Schaeff** [comment]
@m1cm1c can you double check that? I'm still getting "ERROR: Permission to BlockInfinity/ZoKrates-1.git denied to Schaeff" 🤔

**Comment by m1cm1c** [comment]
@Schaeff It's still saying 'Pending Invite'. You probably need to click a link in an Email GitHub sent you to accept that I granted you permission.

**Comment by Schaeff** [comment]
Oops. Done!

**Comment by Schaeff** [comment]
@dark64 can you review this?

File: zokrates_core/src/proof_system/scheme/pghr13.rs
Comment (dark64) [inline zokrates_core/src/proof_system/scheme/pghr13.rs]: I guess `SOLIDITY_G2_ADDITION_LIB` should be removed here, and the usage below in the `format!`
https://github.com/Zokrates/ZoKrates/blob/38f0b4a83a673de390b9da0cdfd21d74b8337fa7/zokrates_core/src/proof_system/scheme/pghr13.rs#L141

**Review by Schaeff — APPROVED** [review]

**Comment by Schaeff** [comment]
Thanks @m1cm1c 👍 !

**Pull request merged**
